docs/tutorials/transforms.ipynb

```diff
@@ -128,56 +168,152 @@
     "            jnp.exp(log_rho)\n",
     "        )\n",
     "\n",
-    "        # Define a custom transform to pass the input coordinates through our `Transformer`\n",
-    "        # network from above\n",
+    "        # Define a custom transform to pass the input coordinates through our\n",
+    "        # `Transformer` network from above\n",
     "        transform = Transformer()\n",
     "        kernel = transforms.Transform(transform, base_kernel)\n",
     "\n",
-    "        # Evaluate and return the GP negative log likelihood as usual\n",
+    "        # Evaluate and return the GP negative log likelihood as usual with the\n",
+    "        # transformed features\n",
     "        gp = GaussianProcess(\n",
     "            kernel, x[:, None], diag=noise**2 + jnp.exp(2 * log_jitter)\n",
     "        )\n",
     "        log_prob, gp_cond = gp.condition(y, t[:, None])\n",
-    "        return -log_prob, (gp_cond.loc, gp_cond.variance)\n",
+    "\n",
+    "        # We return the loss, the conditional mean and variance, and the\n",
+    "        # transformed input parameters\n",
+    "        return (\n",
+    "            -log_prob,\n",
+    "            (gp_cond.loc, gp_cond.variance),\n",
+    "            (transform(x[:, None]), transform(t[:, None])),\n",
```

Comment:
What could be a better way to get the transformed input data, than to return from the `__call__` function in class `DeepLoss`? 🤔 
```python
        return (
            -log_prob,
            (gp_cond.loc, gp_cond.variance),
            (transform(x[:, None]), transform(t[:, None])),
        )
```

I was thinking of replacing it something like this, outside the class functions:
```python
transformer_params = {"params": params_list[0]["params"]["Transformer_0"]}

transformer = Transformer()
x_transform = transformer.apply(transformer_params, x[:, None])
t_transform = transformer.apply(transformer_params, t[:, None])
```

Reply:
I actually think it's fine to do it how you're doing it. Since we `jit` the `value_and_grad` call, I think the extra computations will get optimized out and I think your proposed alternative is a little to fragile (what if the name changes, for example?), so I think it's better to keep it as is!